crud/select/executor.lua

```diff
@@ -110,10 +119,9 @@ function executor.execute(space, index, filter_func, opts)
         local matched, early_exit = filter_func(tuple)
 
         if matched then
-            table.insert(tuples, tuple)
-            tuples_count = tuples_count + 1
+            table.insert(resp.tuples, tuple)
 
-            if opts.limit ~= nil and tuples_count >= opts.limit then
+            if opts.limit ~= nil and #resp.tuples >= opts.limit then
```

Comment:
Using `#` here is possibly not good, see https://gitspartv.github.io/LuaJIT-Benchmarks/#test12.

Reply:
Reworked